example/src/main/java/com/klarna/example/MainActivity.java

```diff
@@ -93,14 +128,86 @@ public void openKlarnaPreferences(View view) {
         Intent intent = new Intent(this, PreferencesActivity.class);
         intent.putExtra(PreferencesActivity.EXTRA_USER_TOKEN, getUserToken());
         startActivityForResult(intent, PREFERENCES_REQUEST_CODE);
+    }
 
+    public void onBuyPressed(View view) {
+        if (this.hasUserToken()) {
+            buyTicket();
+        } else {
+            openKlarnaRegistration();
+        }
     }
+    
+    private void buyTicket() {
+        class purchaseItemRunnable implements Runnable {
```

Comment:
Usually one define inner class instead of the parent (at the end) class and not in a method. WDYT?